### ISSUE TITLE
Manual steps with retention policies causes a NullReferenceException

### ISSUE THREAD

**Issue by graemechristie** [OctopusDeploy/Issues#1701]
Running octopus 3.0.0.1666 - 

I am getting the error 

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Octopus.Shared.Diagnostics.AbstractLog.PlanFutureBlock(String messageText) in y:\work\refs\heads\release\source\Octopus.Shared\Diagnostics\AbstractLog.cs:line 73
```

intermittently when deploying an octopus package for a web (MVC) site, but only when a previous Manual Step is included.

When I deploy to an environment with 4 servers, this issue will generally only occur with 1 of the 4 servers, but that 1 server will be different each time. Re-running the task will usually result in a failure on a different server. 

When I deploy to an environment with a single server, the fault (so far) has happened 100% of the time, but will not occur if I retry the deployment (e.g. from a guided failure).

The error only ever occurs when Step 1, production store Deployment Warning is enabled. if I set it not to run for a deployment, no error occurs.

![image](https://cloud.githubusercontent.com/assets/1361894/8591791/2d6fb8b8-265d-11e5-8d96-b2654a2674d3.png)

The full error message is:

![image](https://cloud.githubusercontent.com/assets/1361894/8591642/4a3bb808-265c-11e5-9460-ffeb759223d5.png)

I'm happy to send you the full logs via email.

**Comment by IanMercer** [comment]
Another thread on this here: http://help.octopusdeploy.com/discussions/problems/36245

**Comment by Dalmirog** [comment]
I wasn't able to reproduce this on 3.0.2.2077 and 3.0.1.2063.

**Comment by PaulStovell** [comment]
This happens when:

 - Retention policies are defined
 - A manual step appears and is proceeded; at this point, the deployment stops, and then recontinues
 - A package step follows the manual step

After the package step, we create a log block for retention policies, but since the deployment paused we lost some information we need for the retention policies log. That's where the null ref comes from. 

Will be fixed in 3.0.5
